03_launch_mgmt_cluster.sh

```diff
@@ -167,57 +295,66 @@ function update_images(){
   done
 }
 
-function kustomize_overlay_bmo() {
-  overlay_path=$1
-cat <<EOF> "$overlay_path/kustomization.yaml"
-apiVersion: kustomize.config.k8s.io/v1beta1
-kind: Kustomization
-configMapGenerator:
-- behavior: merge
-  literals:
-  - PROVISIONING_IP=$CLUSTER_PROVISIONING_IP
-  - PROVISIONING_INTERFACE=$CLUSTER_PROVISIONING_INTERFACE
-  - PROVISIONING_CIDR=$PROVISIONING_CIDR
-  - DHCP_RANGE=$CLUSTER_DHCP_RANGE
-  - DEPLOY_KERNEL_URL=http://$IRONIC_HOST:6180/images/ironic-python-agent.kernel
-  - DEPLOY_RAMDISK_URL=http://$IRONIC_HOST:6180/images/ironic-python-agent.initramfs
-  - IRONIC_ENDPOINT=http://$IRONIC_HOST:6385/v1/
-  - IRONIC_INSPECTOR_ENDPOINT=http://$IRONIC_HOST:5050/v1/
-  - CACHEURL=http://$IRONIC_HOST/images
-  - IRONIC_FAST_TRACK=false
-  name: ironic-bmo-configmap
-resources:
-- $(realpath --relative-to="$overlay_path" "$BMO_CONFIG")
-EOF
-}
-
-
-function deploy_kustomization() {
-    kustomize_overlay_path=$(mktemp -d bmo-XXXXXXXXXX)
-    kustomize_overlay_bmo "$kustomize_overlay_path"
-    pushd "$kustomize_overlay_path"
-
-    # Add custom images in overlay, and override the images with local ones
-    update_kustomization_images
-    popd
-
-    kustomize build "$kustomize_overlay_path" | kubectl apply -f-
-    rm -rf "$kustomize_overlay_path"
-}
-
 function launch_baremetal_operator() {
     pushd "${BMOPATH}"
 
     if [ "${CAPM3_VERSION}" != "v1alpha3" ]; then
       kubectl create namespace metal3
     else
-      BMO_CONFIG="${BMOPATH}/deploy/default"
-      deploy_kustomization
+      # Deploy BMO using deploy.sh script
+
+      # Update container images to use local ones
+      cp "${BMOPATH}/deploy/operator/bmo.yaml" "${BMOPATH}/deploy/operator/bmo.yaml.orig"
+      update_kustomization_images "${BMOPATH}/deploy/operator/bmo.yaml"
+
+      # Update Configmap parameters with correct urls
+      cp "${BMOPATH}/deploy/default/ironic_bmo_configmap.env" "${BMOPATH}/deploy/default/ironic_bmo_configmap.env.orig"
```

Comment:
Can't we replace those vars in place instead of creating a temporary file?

Reply:
the temporary file is to be able to reset it back to what it was, after we have modified it and deployed. So in a way the variables are replaced "in-place" but we have a copy of the file for the reset